lib/spack/spack/build_environment.py

```diff
@@ -371,6 +371,72 @@ def set_module_variables_for_package(pkg, module):
     # Platform-specific library suffix.
     m.dso_suffix = dso_suffix
 
+    def static_to_shared_library(static_lib, shared_lib=None, **kwargs):
+        compiler_path = kwargs.get('compiler', m.spack_cc)
+        compiler = Executable(compiler_path)
+
+        return _static_to_shared_library(pkg.spec.architecture, compiler,
+                                         static_lib, shared_lib, **kwargs)
+
+    m.static_to_shared_library = static_to_shared_library
+
+
+def _static_to_shared_library(arch, compiler, static_lib, shared_lib=None,
+                              **kwargs):
+    """
+    Converts a static library to a shared library.
+
+    Parameters:
+        static_lib (str): Path to the static library.
+        shared_lib (str): Path to the shared library. Default is to derive
+                          from the static library's path.
+
+    Keyword arguments:
+        compiler (str): Path to the compiler. Default is spack_cc.
+        compiler_output: Where to print compiler output to.
+        arguments (str list): Additional arguments for the compiler.
+        version (str): Library version. Default is unspecified.
+    """
+    compiler_output = kwargs.get('compiler_output', None)
+    arguments = kwargs.get('arguments', [])
+    version = kwargs.get('version', None)
+
+    if not shared_lib:
+        shared_lib = '{0}.{1}'.format(os.path.splitext(static_lib)[0],
+                                      dso_suffix)
+
+    compiler_args = []
+
+    if 'linux' in arch:
+        soname = os.path.basename(shared_lib)
+
+        if version:
+            soname += '.{0}'.format(version)
+
+        compiler_args = [
+            '-shared',
```

Comment:
can you add a TODO in here somewhere that says these arguments should really be added to the various compiler classes and generalized for different compilers?